src/Microsoft.AspNetCore.SignalR/HubEndPoint.cs

```diff
@@ -138,15 +139,26 @@ public class HubEndPoint<THub, TClient> : StreamingEndPoint, IInvocationBinder w
             }
         }
 
-        private async Task DispatchMessagesAsync(StreamingConnection connection)
+        private async Task DispatchMessagesAsync(Connection connection)
         {
-            var stream = connection.Transport.GetStream();
             var invocationAdapter = _registry.GetInvocationAdapter(connection.Metadata.Get<string>("formatType"));
 
-            while (true)
+            while (await connection.Transport.Input.WaitToReadAsync())
             {
-                // TODO: Handle receiving InvocationResultDescriptor
-                var invocationDescriptor = await invocationAdapter.ReadMessageAsync(stream, this) as InvocationDescriptor;
+                Message message;
+                if (!connection.Transport.Input.TryRead(out message))
+                {
+                    continue;
```

Comment:
Right, this should never return false (in theory). The code is cleaner this way though as I don't have to nest the entire code block in an if. This should be changed to a loop anyways. I'll update the other PR to loop.